File: algorithms/[7]整数反转/solution.ts
// topics = ["数学"]

function reverse(x: number): number {
  const limit = 2 ** 31;

  let ord = Math.abs(x);
  let now = 0;

  while (ord > 0) {
    now = now * 10 + (ord % 10);
    ord = Math.floor(ord / 10);
  }

  if (x > 0) {
    return now < limit ? now : 0;
  }
  return now <= limit ? 0 - now : 0;
}

File: algorithms/[7]整数反转/solution.py
# topics = ["数学"]

class Solution:
    def reverse(self, x: int) -> int:
        limit = 2 ** 31
        if x > limit:
            return 0

        bool = x > 0
        number = int(str(x if bool else 0 - x)[::-1])

        if number > limit:
            return 0

        return number if bool else 0 - number
